backend/routes/stocks.js

```diff
@@ -10,32 +10,45 @@ router.post("/stocks", async (req, res) => {
         const stock = await Stock.create(req.body);
         res.json({ stock });
     } catch (error) {
-        res.send({ error });
+        res.json({ error });
     }
 });
 
 // Get a specific stock from database
 router.get("/stocks/:ticker", async (req, res) => {
     const ticker = req.params.ticker.toUpperCase();
-    const stock = await Stock.findOne({
-        where: { ticker }
-    });
 
-    if (stock) {
+    try {
+        const stock = await Stock.findOne({
+            where: { ticker }
+        });
+
+        if (stock === null) {
+            return res
+                .status(404)
+                .json({ error: "This stock does not exist in database." });
+        }
+
         const user = req.session.user;
 
-        const StockId = stock.id;
-        const UserId = user.id;
+        if (user != null) {
+            const StockId = stock.id;
+            const UserId = user.id;
 
-        const following = await Follow.findOne({ where: { UserId, StockId } });
+            const following = await Follow.findOne({
+                where: { UserId, StockId }
+            });
 
-        following
-            ? (stock.dataValues.following = true)
-            : (stock.dataValues.following = false);
+            following
+                ? (stock.dataValues.following = true)
+                : (stock.dataValues.following = false);
```

Comment:
Nit: Here the stock object is being mutated. Do you know of a way of avoiding this mutation?